modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/consensus/block/RoundData.scala

```diff
@@ -0,0 +1,48 @@
+package org.tessellation.dag.l1.domain.consensus.block
+
+import cats.syntax.option._
+
+import org.tessellation.dag.domain.block.DAGBlock
+import org.tessellation.dag.l1.domain.block.Tips
+import org.tessellation.dag.l1.domain.consensus.RoundId
+import org.tessellation.dag.l1.domain.consensus.block.BlockConsensusData.{
+  BlockProposal,
+  CancelledBlockCreationRound,
+  Proposal
+}
+import org.tessellation.schema.peer.{Peer, PeerId}
+import org.tessellation.security.signature.Signed
+
+import monocle.macros.syntax.lens._
+
+case class RoundData(
+  roundId: RoundId,
+  peers: Set[Peer],
+  owner: PeerId,
+  ownProposal: Proposal,
+  ownBlock: Option[Signed[DAGBlock]] = None,
+  ownCancellation: Option[CancellationReason] = None,
+  peerProposals: Map[PeerId, Proposal] = Map.empty,
+  peerBlocks: Map[PeerId, Signed[DAGBlock]] = Map.empty,
+  peerCancellations: Map[PeerId, CancellationReason] = Map.empty,
+  tips: Tips
+) {
+
+  def addPeerProposal(proposal: Proposal): RoundData =
+    this.focus(_.peerProposals).modify(_ + (proposal.senderId -> proposal))
+
+  def addOwnBlock(block: Signed[DAGBlock]): RoundData = this.focus(_.ownBlock).replace(block.some)
+
+  def addPeerBlock(blockProposal: BlockProposal): RoundData =
+    this.focus(_.peerBlocks).modify(_ + (blockProposal.senderId -> blockProposal.signedBlock))
+
+  def addOwnCancellation(reason: CancellationReason): RoundData = this.focus(_.ownCancellation).replace(reason.some)
+
+  def addPeerCancellation(cancellation: CancelledBlockCreationRound): RoundData =
+    this.focus(_.peerCancellations).modify(_ + (cancellation.senderId -> cancellation.reason))
+
+  def formBlock(): DAGBlock = DAGBlock(
```

Comment:
I guess it would kind of suggest that we are transforming `RoundData` to a `Block` which is not the case. We are using some of the information stored inside of `RoundData` to form a block. After we form it we are even storing it inside of the `RoundData` again.